codebase/src/java/disco/org/openlvc/disco/application/utils/DrmState.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *   Copyright 2025 Open LVC Project.
+ *
+ *   This file is part of Open LVC Disco.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.openlvc.disco.application.utils;
+
+import java.util.Objects;
+
+import org.openlvc.disco.pdu.record.AngularVelocityVector;
+import org.openlvc.disco.pdu.record.EulerAngles;
+import org.openlvc.disco.pdu.record.VectorRecord;
+import org.openlvc.disco.pdu.record.WorldCoordinate;
+import org.openlvc.disco.utils.Quaternion;
+import org.openlvc.disco.utils.Vec3;
+
+/**
+ * Represents the state of an entity at a point in time in a {@link DeadReckoningModel}.
+ * <p/>
+ * The reference frame of the {@link #velocity} and {@link #acceleration} vectors can vary, and
+ * uses the coordinate system of the model that produced this state.
+ * <p/>
+ * Complete list of reference frames:
+ * <ul>
+ * <li>{@link #position}: world coordinate system (see also: {@link WorldCoordinate})</li>
+ * <li>{@link #velocity}: varies depending on DR model</li>
+ * <li>{@link #acceleration}: varies depending on DR model</li>
+ * <li>{@link #orientation}: world coordinate system (see also: {@link EulerAngles})</li>
+ * <li>{@link #angularVelocity}: entity coordinate system (see also: {@link AngularVelocityVector})</li>
+ * </ul>
+ */
+public record DrmState( Vec3 position,
+                        Vec3 velocity,
+                        Vec3 acceleration,
+                        Quaternion orientation,
+                        Vec3 angularVelocity )
+{
+	//----------------------------------------------------------
+	//                    STATIC VARIABLES
+	//----------------------------------------------------------
+
+	//----------------------------------------------------------
+	//                      CONSTRUCTORS
+	//----------------------------------------------------------
+	public DrmState( WorldCoordinate location, VectorRecord linearVelocity, VectorRecord linearAcceleration, EulerAngles orientation, AngularVelocityVector angularVelocity )
```

Comment:
Same comment as in other places on line wrapping at 100 char width unless it makes things really ugly. I'll stop making it now, but check the whole commit for me.